gluten-core/src/main/scala/org/apache/gluten/utils/PullOutProjectHelper.scala

```diff
@@ -151,6 +183,22 @@ trait PullOutProjectHelper {
       case ae: AggregateExpression => rewriteAggregateExpression(ae, expressionMap)
       case other => other
     }
-    we.copy(windowFunction = newWindowFunc)
+
+    val newWindowSpec = we.windowSpec.frameSpecification match {
+      case swf: SpecifiedWindowFrame if needPreComputeRangeFrame(swf) =>
+        // This is guaranteed by Spark, but we still check it here
+        if (orderSpecs.size != 1) {
+          throw new GlutenNotSupportException(
```

Comment:
Let's use `GlutenException`. Generally `GlutenNotSupportException` is caught in fallback handling for some limitation cases. But for this case, I think we can directly let a runtime exception happen.

Reply:
done